packages/axelar-wasm-std/src/address.rs

```diff
@@ -54,14 +54,27 @@ pub fn validate_address(address: &str, format: &AddressFormat) -> Result<(), Err
                 bail!(Error::InvalidAddress("0x prefix is missing".to_string()))
             }
 
+            let felt = Felt::from_hex(address)
+                .change_context(Error::InvalidAddress(address.to_string()))?;
+
+            // NOTE: I don't like this comparison, but the Felt struct, doesn't error if the value
+            // that get passed exceeds Felt::MAX. It just changes the value to a number close to
+            // Felt::MAX (0x07fffffffffffdf0ffffffffffffffffffffffffffffffffffffffffffffffe0).
+            //
+            // So we have no way to check if the passed adderss overflows Felt, other than checking
+            // that the passed value is different than the parsed felt.
+            if format!("0x{:064x}", felt) != address {
```

Comment:
@puhtaytow 
I really dislike this part, but Felt overflow was the main reason I did this custom address format